run.py

```diff
@@ -225,8 +240,33 @@ def reindex(since=None, batch_size=None, force=False, update_solr=True, update_m
         diagnostics(args.bibcodes)
         
     logger.info(args)
-    
-    if args.reindex:
+
+    if args.validate:
+        fields = ('abstract', 'ack', 'aff', 'alternate_bibcode', 'alternate_title', 'arxiv_class', 'author',
+                  'author_count', 'author_facet', 'author_facet_hier', 'author_norm', 'bibgroup', 'bibgroup_facet',
+                  'bibstem', 'bibstem_facet', 'body', 'citation', 'citation_count', 'cite_read_boost', 'classic_factor',
+                  'comment', 'copyright', 'data', 'data_count', 'data_facet', 'database', 'date', 'doctype',
+                  'doctype_facet_hier',
+                  'doi', 'eid', 'email', 'entry_date', 'esources', 'first_author', 'first_author_facet_hier',
+                  'first_author_norm', 'fulltext_mtime', 'grant', 'grant_facet_hier', 'id', 'identifier', 'indexstamp',
+                  'isbn', 'issn', 'issue', 'keyword', 'keyword_facet', 'keyword_norm', 'keyword_schema', 'lang',
+                  'links_data', 'metadata_mtime', 'metrics_mtime', 'nedid', 'nedtype', 'ned_object_facet_hier',
+                  'nonbib_mtime',
+                  'origin', 'orcid_mtime', 'orcid', 'orcid_pub', 'orcid_user', 'orcid_other', 'page', 'page_range',
+                  'page_count', 'property', 'pub', 'pub_raw', 'pubdate', 'pubnote', 'read_count', 'reader', 'recid',
+                  'reference', 'simbad_object_facet_hier', 'simbid', 'simbtype', 'title', 'update_timestamp', 'vizier',
+                  'vizier_facet', 'volume', 'year')
+
+        ignore_fields = ('id', 'indexstamp', 'fulltext_mtime', 'links_data', 'metadata_mtime', 'metrics_mtime',
+                         'nonbib_mtime', 'orcid_mtime', 'recid', 'update_timestamp')
+
+        new_fields = ('data_count', 'entry_date', 'esources', 'nedid', 'nedtype', 'ned_object_facet_hier', 'origin',
+                      'page_count', 'page_range')
+
+        d = validate.Validate(fields, ignore_fields, new_fields)
```

Comment:
this is forward thinking, like it!

defaults should be part of the validate module: validate.Validate() -- no hard/fast rule, only that it will be self contained (and run.py doensn't offer an option to change the fields)

you can make that change in some next iteration, I think there will be some - and the validate.py should also have its own unittest